src/App.tsx

```diff
@@ -8,37 +8,117 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUsers } from './services/users';
+import { getPosts } from './services/userPost';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [arePostLoading, setArePostLoading] = useState(false);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers = await getUsers();
+
+      setUsers(loadedUsers);
+    } catch {
+      throw new Error('Failed to load users');
```

Comment:
better not to throw an error - it'll lead to a page crash (you do not catch and handle the error which you throw here somewhere else) - FYI

Reply:
can you give some example, how to do it in the best way(-s)?